### PR TITLE
i18n(zh-cn): translate 404、environmental-impact、project-structure、showcase

### PR DESCRIPTION
**Pull request by liruifengv**
#### What kind of changes does this PR include?

- Changes or translations Starlight docs site content

#### Description

-  translate 404、environmental-impact、project-structure、showcase

### CODE REVIEW

File: docs/src/content/docs/zh/404.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+title: 页面未找到
+template: splash
+editUrl: false
+hero:
+  title: '404'
+  tagline: <strong>休斯顿，我们遇到了一个问题。</strong>我们找不到该页面。<br>请检查 URL 或尝试使用搜索栏。
+  actions:
+    - text: 返回首页
+      icon: right-arrow
+      link: /
```
Comment (delucis) [inline docs/src/content/docs/zh/404.md:11]: We should link to the Chinese-language homepage here:

```suggestion
      link: /zh/
```

File: docs/src/content/docs/zh/showcase.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+---
+title: Starlight Showcase
```
Comment (delucis) [inline docs/src/content/docs/zh/showcase.md:2]: Noticing that this isn’t translated here or in the sidebar config:
image
Is that intentional?

File: docs/src/content/docs/zh/environmental-impact.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+- [“Building a greener web”](https://youtu.be/EfPoOt7T5lg)，Michelle Barker 的演讲
+- [“Sustainable Web Development Strategies Within An Organization”](https://www.smashingmagazine.com/2022/10/sustainable-web-development-strategies-organization/)，Michelle Barker 的文章
+- [“A sustainable web for everyone”](https://2021.stateofthebrowser.com/speakers/tom-greenwood/)，Tom Greenwood 的演讲
+- [“How Web Content Can Affect Power Usage”](https://webkit.org/blog/8970/how-web-content-can-affect-power-usage/)，Benjamin Poulain 和 Simon Fraser 的文章
+
+[sf]: https://www.sciencefocus.com/science/what-is-the-carbon-footprint-of-the-internet/
+[bbc]: https://www.bbc.com/future/article/20200305-why-your-internet-habits-are-not-as-clean-as-you-think
+[http]: https://httparchive.org/reports/state-of-the-web
+[assets]: https://docs.astro.build/en/guides/assets/
+[islands]: https://docs.astro.build/en/concepts/islands/
```
Comment (delucis) [inline docs/src/content/docs/zh/environmental-impact.md:139]: We can link to the translated docs here!

```suggestion
[assets]: https://docs.astro.build/zh-cn/guides/assets/
[islands]: https://docs.astro.build/zh-cn/concepts/islands/
```

File: docs/src/content/docs/zh/guides/project-structure.mdx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+---
+title: 项目结构
+description: 了解如何组织 Starlight 项目中的文件。
+---
+
+本指南将向你展示 Starlight 项目的组织方式以及项目中的不同文件的作用。
+
+Starlight 项目通常遵循与其他 Astro 项目相同的文件和目录结构。有关更多详细信息，请参阅 [Astro 的项目结构文档](https://docs.astro.build/en/core-concepts/project-structure/)。
```
Comment (delucis) [inline docs/src/content/docs/zh/guides/project-structure.mdx:8]: Same here:

```suggestion
Starlight 项目通常遵循与其他 Astro 项目相同的文件和目录结构。有关更多详细信息，请参阅 [Astro 的项目结构文档](https://docs.astro.build/zh-cn/core-concepts/project-structure/)。
```

**Review by delucis — COMMENTED** [review]
Thank you for these new pages @liruifengv! I spotted a few small details.

File: docs/src/content/docs/zh/showcase.md (hunk 2 above)
Comment (liruifengv) [reply docs/src/content/docs/zh/showcase.md:2]: Yes, I don't think `showcase` needs to be translated. In China, we also use the word `showcase` in our work.

**Review by delucis — APPROVED** [review]
Perfect, thank you @liruifengv 🚀

**Pull request merged**